test/extended/pod_autoscaling/deployment.go

```diff
@@ -0,0 +1,148 @@
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	g.It("should only scale the most recent deployment", func() {
+		err := oc.Run("deploy").Args(dcName, "--latest").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		rcClient := oc.KubeREST().ReplicationControllers(oc.Namespace())
+		err = exutil.WaitForADeployment(rcClient, dcName, exutil.CheckDeploymentCompletedFunc, exutil.CheckDeploymentFailedFunc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		err = oc.Run("deploy").Args(dcName, "--latest").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		err = exutil.WaitForADeployment(rcClient, dcName, exutil.CheckDeploymentCompletedFunc, exutil.CheckDeploymentFailedFunc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("creating a new HPA targeted at 80% CPU consumption")
+		scaler := exapi.HorizontalPodAutoscaler{
+			ObjectMeta: kapi.ObjectMeta{Name: "frontend-scaler"},
+			Spec: exapi.HorizontalPodAutoscalerSpec{
+				ScaleRef: exapi.SubresourceReference{
+					Kind: "DeploymentConfig",
+					Name: dcName,
+					APIVersion: "v1",
+					Subresource: "scale",
+				},
+				MaxReplicas: 6,
+				CPUUtilization: &exapi.CPUTargetUtilization{TargetPercentage: 80},
+			},
+		}
+		_, err = oc.KubeREST().Extensions().HorizontalPodAutoscalers(oc.Namespace()).Create(&scaler)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		err = waitForRCToBeScaled(rcClient, dcName+"-1", 0)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		err = waitForRCToBeScaled(rcClient, dcName+"-2", 1)
+		o.Expect(err).NotTo(o.HaveOccurred())
+	})
+
+	g.It("should not scale a failed or incomplete deployment", func() {
+		err := oc.Run("deploy").Args(dcName, "--latest").Execute()
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		rcClient := oc.KubeREST().ReplicationControllers(oc.Namespace())
+		err = exutil.WaitForADeployment(rcClient, dcName, exutil.CheckDeploymentCompletedFunc, exutil.CheckDeploymentFailedFunc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// update the deployment to mark it as failed
+		rc, err := rcClient.Get(dcName+"-1")
+		o.Expect(err).NotTo(o.HaveOccurred())
+		rc.Annotations[deployapi.DeploymentStatusAnnotation] = string(deployapi.DeploymentStatusFailed)
+		_, err = rcClient.Update(rc)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		g.By("creating a new HPA targeted at 80% CPU consumption")
+		scaler := exapi.HorizontalPodAutoscaler{
+			ObjectMeta: kapi.ObjectMeta{Name: "frontend-scaler"},
+			Spec: exapi.HorizontalPodAutoscalerSpec{
+				ScaleRef: exapi.SubresourceReference{
+					Kind: "DeploymentConfig",
+					Name: dcName,
+					APIVersion: "v1",
+					Subresource: "scale",
+				},
+				MaxReplicas: 6,
+				CPUUtilization: &exapi.CPUTargetUtilization{TargetPercentage: 80},
+			},
+		}
+		_, err = oc.KubeREST().Extensions().HorizontalPodAutoscalers(oc.Namespace()).Create(&scaler)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		// this needs to be "wait till RC *could* be scaled (check last scale time?)
+		err = waitForRCToBeScaled(rcClient, dcName+"-1", 1)
```

Comment:
The autoscaler scales once every certain interval.  We need to wait at least until the scaler could have scaled at least once.

Otherwise, we could check the scale, see that it hasn't changed, say "ok, we're good", and then 1 minute later the scaler actually scales it down (assuming the fail logic is broken).